circuit/src/execution-delivery/src/lib.rs

```diff
@@ -795,7 +795,9 @@ impl<T: Config> Pallet<T> {
     }
 
     pub fn select_authority(escrow_account: T::AccountId) -> Result<AuthorityId, &'static str> {
-        let local_keys = AuthorityId::all();
+        let mut local_keys = AuthorityId::all();
+
+        local_keys.sort();
```

Comment:
Fix looks fine but sorting the AuthorityId array during runtime will heavily impact weight of execution delivery as we scale upwards. At some future point we should definitely look for a way to make sure our authority ids are always sorted before invoking the `select_authority` function.